File: Lab №4/C++/TTriangle.cpp
#include <sstream>
#include "TTriangle.h"

TTriangle::TTriangle(Coordinates point1, Coordinates point2, int side1,  int angle)
{
	coordinates.push_back(point1);
	coordinates.push_back(point2);
	double x3 = 0.0;
	double y3 = 0.0;
	double side3 = point1.distanceTo(point2);
	double side2 = sqrt(pow(side1, 2) + pow(side3, 2) - 2 * side1 * side3 * cos((angle * 3.14) / 180));

	double k = (pow(side3, 2) + pow(side2, 2) - pow(side1, 2)) / (2 * side3);   //обчислення координат третьої вершини
	double h = sqrt(pow(side2, 2) - pow(k, 2));
	if (angle <= 90) {
		x3 = round((point1.x + (k / side3) * (point2.x - point1.x) - (h / side3) * (point2.y - point1.y)) * 100) / 100;
		y3 = round((point1.y + (k / side3) * (point2.y - point1.y) + (h / side3) * (point2.x - point1.x)) * 100) / 100;
	}
	else {
		x3 = round((point1.x + (k / side3) * (point2.x - point1.x) + (h / side3) * (point2.y - point1.y)) * 100) / 100;
		y3 = round((point1.y + (k / side3) * (point2.y - point1.y) - (h / side3) * (point2.x - point1.x)) * 100) / 100;
	}
	Coordinates point3(x3, y3);
	coordinates.push_back(point3);
}

//string TTriangle::info()
//{
//	stringstream output;
//	int i = 1;
//	for (auto point:coordinates)
//	{
//		output << "x" << i << " = " << point.x << "; y" << i << " = " << point.y << endl;
//	}
//	output << "side1 = ";
//	return output.str();
//}

File: Lab №4/Python/TTriangle.py
import math


def equilateral(triangle):
    area = triangle.calculate_area()
    print(triangle.info())
    return area


def right(triangle):
    area = triangle.calculate_area()
    print(triangle.info())
    return area


def isosceles(triangle):
    perimeter = triangle.calculate_perimeter()
    print(triangle.info())
    return perimeter


# словник для визначення виду трикутника
TriangleType = {
    'Equilateral': equilateral,
    'Right': right,
    'Isosceles': isosceles
}


class Coordinates:

    def __init__(self, x, y):
        self.x = x
        self.y = y

    def distance_to(self, point):
        return math.sqrt((point.x - self.x) ** 2 + (point.y - self.y) ** 2)


class TTriangle:

    def __init__(self, point1, point2, side1, angle):
        self.angle = angle
        self.side1 = side1
        self.coordinates_list = []
        self.coordinates_list.append(point1)
        self.coordinates_list.append(point2)
        self.side3 = point1.distance_to(point2)
        self.side2 = math.sqrt(side1 ** 2 + self.side3 ** 2 - 2 * side1 * self.side3 * math.cos((angle * 3.14) / 180))

        # обчислення координат третьої вершини
        k = (self.side3 ** 2 + self.side2 ** 2 - side1 ** 2) / (2 * self.side3)
        h = math.sqrt(self.side2 ** 2 - k ** 2)
        if angle <= 90:
            x3 = round((point1.x + (k / self.side3) * (point2.x - point1.x) - (h / self.side3) * (point2.y - point1.y)), 2)
            y3 = round((point1.y + (k / self.side3) * (point2.y - point1.y) + (h / self.side3) * (point2.x - point1.x)), 2)
        else:
            x3 = round((point1.x + (k / self.side3) * (point2.x - point1.x) + (h / self.side3) * (point2.y - point1.y)), 2)
            y3 = round((point1.y + (k / self.side3) * (point2.y - point1.y) - (h / self.side3) * (point2.x - point1.x)), 2)
        point3 = Coordinates(x3, y3)
        self.coordinates_list.append(point3)

    def calculate_area(self):
        return 0

    def calculate_perimeter(self):
        return 0

    def info(self):
        return 0

